graphbolt/src/unique_and_compact.h

```diff
@@ -0,0 +1,60 @@
+/**
+ *  Copyright (c) 2023 by Contributors
+ *
+ * @file unique_and_compact.h
+ * @brief Unique and compact op.
+ */
+#ifndef GRAPHBOLT_SHM_UTILS_H_
+#define GRAPHBOLT_SHM_UTILS_H_
+
+#include <torch/torch.h>
+
+namespace graphbolt {
+namespace sampling {
+
+using TensorList = std::vector<torch::Tensor>;
+/**
+ * @brief Removes duplicate elements from the concatenated 'unique_dst_ids' and
+ * 'src_ids' tensors and applies the uniqueness information to compact both
+ * source and destination tensor lists.
+ *
+ * The function performs two main operations:
+ *   1. Unique Operation: Concatenates 'unique_dst_ids' and 'src_ids' tensors
+ * and extracts unique elements, ensuring no duplicates. The resulting tensor
+ * represents all unique elements in both 'src_ids' and 'dst_ids', and the
+ * indices correspond to the compacted IDs of the corresponding elements.
```

Comment:
1. Unique Operation: unique(concat(unique_dst_ids, src_ids)), in which the unique operator will guarantee the unique_dst_ids are at the head of the result tensor.